pkg/specgen/generate/container_create.go

```diff
@@ -238,6 +238,17 @@ func createContainerOptions(ctx context.Context, rt *libpod.Runtime, s *specgen.
 	if s.Entrypoint != nil {
 		options = append(options, libpod.WithEntrypoint(s.Entrypoint))
 	}
+	// If the user did not set an entrypoint but the image did, ensure it is
+	// created.
+	if s.Entrypoint == nil && img != nil {
+		newEntry, err := img.Entrypoint(ctx)
+		if err != nil {
+			return nil, err
+		}
+		if len(newEntry) > 0 {
+			options = append(options, libpod.WithCreateWorkingDir())
+		}
```

Comment:
shouldn't we look at `img.WorkingDir(...)`?  Why does it depend on the entrypoint?

Reply:
Oops, fixed